Replace Module::copy_into with Module::clone.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24068 Replace Module::copy_into with Module::clone.**

The new method has a simpler interface (no arguments).

Resolves #23915.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24068

Differential Revision: [D16736379](https://our.internmc.facebook.com/intern/diff/D16736379)

Commit:
Replace Module::copy_into with Module::clone.

The new method has a simpler interface (no arguments).

Commit:
Fix clang-tidy on "Replace Module::copy_into with Module::clone."


The new method has a simpler interface (no arguments).

zdevito reviewed (approved):
Looks good, much clearer!

Commit:
Rebase on "Replace Module::copy_into with Module::clone."


The new method has a simpler interface (no arguments).

Differential Revision: [D16736379](https://our.internmc.facebook.com/intern/diff/D16736379)

Commit:
Update on "Replace Module::copy_into with Module::clone."


The new method has a simpler interface (no arguments).

Resolves #23915.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24068

Differential Revision: [D16736379](https://our.internmc.facebook.com/intern/diff/D16736379)

Commit:
Update on "Replace Module::copy_into with Module::clone."


The new method has a simpler interface (no arguments).

Resolves #23915.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24068

Differential Revision: [D16736379](https://our.internmc.facebook.com/intern/diff/D16736379)

Commit:
Rebase on "Replace Module::copy_into with Module::clone."


The new method has a simpler interface (no arguments).

Resolves #23915.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24068

Differential Revision: [D16736379](https://our.internmc.facebook.com/intern/diff/D16736379)
